Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using NServiceBus;
+using NServiceBus.Azure.Transports.WindowsAzureStorageQueues;
+
+class CustomEnvelopeUnwrapper
+{
+
+    CustomEnvelopeUnwrapper(EndpointConfiguration endpointConfiguration)
+    {
+        #region CustomEnvelopeUnwrapper
+
+        var transport = endpointConfiguration.UseTransport<AzureStorageQueueTransport>();
+
+        transport.UnwrapMessagesWith(cloudQueueMessage =>
+        {
+            using (var stream = new MemoryStream(cloudQueueMessage.AsBytes))
+            using (var streamReader = new StreamReader(stream))
+            using (var textReader = new JsonTextReader(streamReader))
+            {
+                //try deserialize to a NServiceBus envelope first
+                var wrapper = jsonSerializer.Deserialize<MessageWrapper>(textReader);
+
+                if (!string.IsNullOrEmpty(wrapper.Id))
```

Comment:
Empty isn't valid so we don't return? (are you saying this should be `!= null` ?)